<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Getting Started with glyparse</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Getting Started with glyparse</h1>



<div id="your-universal-glycan-text-translator" class="section level2">
<h2>Your Universal Glycan Text Translator üîÑ</h2>
<p>Welcome to the world of glycan text parsing! If you‚Äôve ever worked with glycan data from different sources, you know the frustration: every database, software tool, and research group seems to have their own way of representing glycan structures in text format.</p>
<p>That‚Äôs where <code>glyparse</code> comes to the rescue! üöÄ</p>
<p>Think of <code>glyparse</code> as your <strong>universal glycan translator</strong> ‚Äî it can read glycan structures written in many different ‚Äúlanguages‚Äù and convert them all into a unified format that your computer can understand and work with.</p>
<p><strong>Note:</strong> All functions in <code>glyparse</code> return <code>glyrepr::glycan_structure</code> objects. If you are unfamiliar with <code>glyrepr</code>, you can read the documentation <a href="https://glycoverse.github.io/glyrepr/articles/glyrepr.html">here</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glyparse)</span></code></pre></div>
</div>
<div id="the-babel-tower-of-glycan-text-formats" class="section level2">
<h2>The Babel Tower of Glycan Text Formats üóº</h2>
<p>Before we dive in, let‚Äôs see what we‚Äôre dealing with. Here‚Äôs the same N-glycan core structure written in different formats:</p>
<table>
<colgroup>
<col width="22%" />
<col width="25%" />
<col width="52%" />
</colgroup>
<thead>
<tr class="header">
<th>Format</th>
<th>Example</th>
<th>Where You‚Äôll See It</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>IUPAC-condensed</strong></td>
<td><code>Man(a1-3)[Man(a1-6)]Man(b1-4)GlcNAc(b1-4)GlcNAc</code></td>
<td>Literature, UniCarbKB</td>
</tr>
<tr class="even">
<td><strong>IUPAC-short</strong></td>
<td><code>Mana3(Mana6)Manb4GlcNAcb4GlcNAc</code></td>
<td>Literature, UniCarbKB</td>
</tr>
<tr class="odd">
<td><strong>IUPAC-extended</strong></td>
<td><code>alpha-D-Man-(1-&gt;3)-[alpha-D-Man-(1-&gt;6)]-beta-D-Man-(1-&gt;4)-beta-D-GlcNAc-(1-&gt;4)-D-GlcNAc</code></td>
<td>Literature, UniCarbKB</td>
</tr>
<tr class="even">
<td><strong>GlycoCT</strong></td>
<td>Complex multi-line format</td>
<td>Literature, GlycomeDB</td>
</tr>
<tr class="odd">
<td><strong>WURCS</strong></td>
<td><code>WURCS=2.0/3,5,4/[...]/1-1-2-3-3/a4-b1_b4-c1...</code></td>
<td>Literature, GlyTouCan</td>
</tr>
<tr class="even">
<td><strong>Linear Code</strong></td>
<td><code>Ma3(Ma6)Mb4GNb4GNb</code></td>
<td>Literature</td>
</tr>
<tr class="odd">
<td><strong>pGlyco</strong></td>
<td><code>(N(N(H(H(H)))))</code></td>
<td>pGlyco software results</td>
</tr>
<tr class="even">
<td><strong>StrucGP</strong></td>
<td><code>A2B2C1D1E2fedcba</code></td>
<td>StrucGP software results</td>
</tr>
</tbody>
</table>
<p>Confusing, right? üòµ‚Äçüí´ <code>glyparse</code> understands them all!</p>
</div>
<div id="your-parsing-toolkit" class="section level2">
<h2>Your Parsing Toolkit üõ†Ô∏è</h2>
<p><code>glyparse</code> provides seven specialized parsers, each optimized for a specific format:</p>
<ul>
<li><strong><code>parse_iupac_condensed()</code></strong>: The most common format</li>
<li><strong><code>parse_iupac_short()</code></strong>: Compact literature format<br />
</li>
<li><strong><code>parse_iupac_extended()</code></strong>: Verbose formal format</li>
<li><strong><code>parse_glycoct()</code></strong>: Database standard format</li>
<li><strong><code>parse_wurcs()</code></strong>: Modern standardized format</li>
<li><strong><code>parse_linear_code()</code></strong>: Linear Code format</li>
<li><strong><code>parse_pglyco_struc()</code></strong>: pGlyco software format</li>
<li><strong><code>parse_strucgp_struc()</code></strong>: StrucGP software format</li>
</ul>
<p>All parsers follow the same pattern:</p>
<ul>
<li><strong>Input</strong>: Character vector of structure strings</li>
<li><strong>Output</strong>: A <code>glyrepr::glycan_structure</code> object that you can analyze</li>
</ul>
</div>
<div id="part-0-auto_parse" class="section level2">
<h2>Part 0: <code>auto_parse()</code></h2>
<p>Don‚Äôt know what you‚Äôre dealing with? Give it to <code>auto_parse()</code>! This function tries to identify the format automatically and use the appropriate parser. Even input with mixed formats is supported.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Gal(b1-3)GalNAc(b1-&quot;</span>,</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;(N(F)(N(H(H(N))(H(N(H))))))&quot;</span>,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;WURCS=2.0/3,3,2/[a2122h-1b_1-5][a1122h-1b_1-5][a1122h-1a_1-5]/1-2-3/a4-b1_b3-c1&quot;</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">auto_parse</span>(x)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;glycan_structure[3]&gt;</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] Gal(b1-3)GalNAc(b1-</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [2] Hex(??-?)HexNAc(??-?)Hex(??-?)[HexNAc(??-?)Hex(??-?)]Hex(??-?)HexNAc(??-?)[dHex(??-?)]HexNAc(??-</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [3] Man(a1-3)Man(b1-4)Glc(b1-</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Unique structures: 3</span></span></code></pre></div>
</div>
<div id="part-1-iupac-family-the-popular-kids" class="section level2">
<h2>Part 1: IUPAC Family ‚Äî The Popular Kids üåü</h2>
<p>Let‚Äôs start with the IUPAC formats.</p>
<div id="iupac-condensed-the-literature-standard" class="section level3">
<h3>IUPAC-Condensed: The Literature Standard</h3>
<p>This format is widely used in scientific literature and databases like UniCarbKB.</p>
<p>Want to know more about IUPAC-condensed format? Check <a href="https://glycoverse.github.io/glyrepr/articles/iupac.html">this</a> out!</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Single structure</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>iupac_condensed <span class="ot">&lt;-</span> <span class="st">&quot;Neu5Ac(a2-3)Gal(b1-4)[Fuc(a1-3)]GlcNAc(b1-4)Gal(b1-4)Glc(a1-&quot;</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">parse_iupac_condensed</span>(iupac_condensed)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;glycan_structure[1]&gt;</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] Neu5Ac(a2-3)Gal(b1-4)[Fuc(a1-3)]GlcNAc(b1-4)Gal(b1-4)Glc(a1-</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Unique structures: 1</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Multiple structures at once</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>glycans <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Man(a1-3)[Man(a1-6)]Man(b1-4)GlcNAc(b1-4)GlcNAc(b1-&quot;</span>,  <span class="co"># N-glycan core</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Gal(b1-3)GalNAc(b1-&quot;</span>,                                  <span class="co"># O-glycan core 1</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Neu5Ac(a2-3)Gal(b1-3)[GlcNAc(b1-6)]GalNAc(b1-&quot;</span>         <span class="co"># O-glycan core 2</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">parse_iupac_condensed</span>(glycans)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;glycan_structure[3]&gt;</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] Man(a1-3)[Man(a1-6)]Man(b1-4)GlcNAc(b1-4)GlcNAc(b1-</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [2] Gal(b1-3)GalNAc(b1-</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [3] Neu5Ac(a2-3)Gal(b1-3)[GlcNAc(b1-6)]GalNAc(b1-</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Unique structures: 3</span></span></code></pre></div>
</div>
<div id="iupac-short-literatures-favorite" class="section level3">
<h3>IUPAC-Short: Literature‚Äôs Favorite</h3>
<p>This compact format is popular in research papers because it saves space:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># The same structures in short format</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>iupac_short <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Mana3(Mana6)Manb4GlcNAcb4GlcNAcb-&quot;</span>,</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Galb3GalNAcb-&quot;</span>, </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Neu5Aca3Galb3(GlcNAcb6)GalNAcb-&quot;</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">parse_iupac_short</span>(iupac_short)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;glycan_structure[3]&gt;</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] Man(a1-3)[Man(a1-6)]Man(b1-4)GlcNAc(b1-4)GlcNAc(b1-</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [2] Gal(b1-3)GalNAc(b1-</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [3] Neu5Ac(a2-3)Gal(b1-3)[GlcNAc(b1-6)]GalNAc(b1-</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Unique structures: 3</span></span></code></pre></div>
<p>Notice how much more compact this is! The parser is smart enough to infer common linkage positions (like Neu5Ac always being a2-linked).</p>
</div>
<div id="iupac-extended-the-formal-one" class="section level3">
<h3>IUPAC-Extended: The Formal One</h3>
<p>This verbose format includes full chemical names and stereochemistry:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>iupac_extended <span class="ot">&lt;-</span> <span class="fu">paste0</span>(</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Œ±-D-Manp-(1‚Üí3)[Œ±-D-Manp-(1‚Üí6)]-Œ≤-D-Manp-(1‚Üí4)&quot;</span>,</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;-Œ≤-D-GlcpNAc-(1‚Üí4)-Œ≤-D-GlcpNAc-(1‚Üí&quot;</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">parse_iupac_extended</span>(iupac_extended)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;glycan_structure[1]&gt;</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] Man(a1-3)[Man(a1-6)]Man(b1-4)GlcNAc(b1-4)GlcNAc(b1-</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Unique structures: 1</span></span></code></pre></div>
</div>
</div>
<div id="part-2-database-formats-the-heavy-hitters" class="section level2">
<h2>Part 2: Database Formats ‚Äî The Heavy Hitters üí™</h2>
<div id="glycoct-the-precision-format" class="section level3">
<h3>GlycoCT: The Precision Format</h3>
<p>GlycoCT is used in literature for precise representation and in databases like GlycomeDB. It‚Äôs more complex but extremely precise:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>glycoct <span class="ot">&lt;-</span> <span class="fu">paste0</span>(</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;RES</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;1b:b-dglc-HEX-1:5</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;2b:b-dgal-HEX-1:5</span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;3b:a-dgal-HEX-1:5</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;LIN</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;1:1o(4+1)2d</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;2:2o(3+1)3d&quot;</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="fu">parse_glycoct</span>(glycoct)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;glycan_structure[1]&gt;</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] Gal(a1-3)Gal(b1-4)Glc(b1-</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Unique structures: 1</span></span></code></pre></div>
</div>
<div id="wurcs-the-complex-structure-format" class="section level3">
<h3>WURCS: The Complex Structure Format</h3>
<p>WURCS (Web3 Unique Representation of Carbohydrate Structures) is used in literature for complex structures and in databases like GlyTouCan:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>wurcs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;WURCS=2.0/3,3,2/&quot;</span>,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;[a2122h-1b_1-5][a1122h-1b_1-5][a1122h-1a_1-5]/&quot;</span>,</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;1-2-3/a4-b1_b3-c1&quot;</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">parse_wurcs</span>(wurcs)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;glycan_structure[1]&gt;</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] Man(a1-3)Man(b1-4)Glc(b1-</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Unique structures: 1</span></span></code></pre></div>
</div>
<div id="linear-code-the-simplified-format" class="section level3">
<h3>Linear Code: The Simplified Format</h3>
<p>Linear Code is a simplified format used in literature for complex structures:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>linear_code <span class="ot">&lt;-</span> <span class="st">&quot;Ma3(Ma6)Mb4GNb4GNb&quot;</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">parse_linear_code</span>(linear_code)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;glycan_structure[1]&gt;</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] Man(a1-3)[Man(a1-6)]Man(b1-4)GlcNAc(b1-4)GlcNAc(b1-</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Unique structures: 1</span></span></code></pre></div>
</div>
</div>
<div id="part-3-software-specific-formats-the-specialists" class="section level2">
<h2>Part 3: Software-Specific Formats ‚Äî The Specialists üî¨</h2>
<div id="pglyco-format-proteomics-tool" class="section level3">
<h3>pGlyco Format: Proteomics Tool</h3>
<p>If you work with glycoproteomics, you might encounter pGlyco‚Äôs parenthetical notation:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>pglyco <span class="ot">&lt;-</span> <span class="st">&quot;(N(F)(N(H(H(N))(H(N(H))))))&quot;</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">parse_pglyco_struc</span>(pglyco)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;glycan_structure[1]&gt;</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] Hex(??-?)HexNAc(??-?)Hex(??-?)[HexNAc(??-?)Hex(??-?)]Hex(??-?)HexNAc(??-?)[dHex(??-?)]HexNAc(??-</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Unique structures: 1</span></span></code></pre></div>
<p>This cryptic notation actually represents a complex N-glycan:</p>
<ul>
<li>N = HexNAc</li>
<li>F = Fuc<br />
</li>
<li>H = Hex (Man or Gal)</li>
</ul>
</div>
<div id="strucgp-format-alphabetical-system" class="section level3">
<h3>StrucGP Format: Alphabetical System</h3>
<p>StrucGP uses a letter-based encoding system:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>strucgp <span class="ot">&lt;-</span> <span class="st">&quot;A2B2C1D1E2F1fedD1E2edcbB5ba&quot;</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">parse_strucgp_struc</span>(strucgp)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;glycan_structure[1]&gt;</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] Hex(??-?)HexNAc(??-?)Hex(??-?)[HexNAc(??-?)Hex(??-?)]Hex(??-?)HexNAc(??-?)[dHex(??-?)]HexNAc(??-</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Unique structures: 1</span></span></code></pre></div>
</div>
</div>
<div id="the-bottom-line" class="section level2">
<h2>The Bottom Line üéØ</h2>
<p><code>glyparse</code> transforms the chaos of glycan text formats into order. No matter where your glycan data comes from, databases, literature, or software tools, you can now parse it into <code>glyrepr::glycan_structure()</code> for further analysis. In fact, <code>glyread</code> package uses these parsing functions internally when reading output from common glycopeptide identification softwares.</p>
<p><strong>Next steps:</strong></p>
<ul>
<li>Explore the <code>glyrepr</code> package for structure manipulation</li>
<li>Try <code>glymotif</code> for motif analysis of your parsed structures<br />
</li>
<li>Use <code>glyexp</code> for experimental data analysis</li>
<li>Check out the rest of the <code>glycoverse</code> ecosystem!</li>
</ul>
<p>Happy parsing! üß¨‚ú®</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
